### PR TITLE
Flagging computed fields in ORM

### PR DESCRIPTION
**Pull request by larsakerson**
Add flag for computed fields in `orm.fields.Field` and use this attribute to resolve error on record updates in `orm.model.Model.save()`

### CODE REVIEW

**Comment by gtalarico** [comment]
Thanks for the quick turn around.

This looks good to me. 
Just need a few more things to make it ready to ship:

* Can you add a unit test in the orm test suite?
* maybe a note in the orm docs?

**Comment by NicoHood** [comment]
Wouldnt it make sense to rename this parameter to something else? I was thinking about this, as I need to download images via airtable:
https://github.com/gtalarico/pyairtable/issues/179

Now if we want to add this image implementation, I think it could be tricky to get it right to support download and upload. I think for the first run a download-only option would make sense, and then this PR comes into play:

In this case the image fields are not computed, but we rather want to treat it `read-only`. So maybe a more generic name would be `read_only` in this case?

**Comment by aleafa** [comment]
Your suggestion makes sense to me, @NicoHood. I've updated the parameter name to `read_only`.

@gtalarico I'm not very familiar with pytest 😬 but I added a test and updated the test suite for this feature and they pass. Might be worth another set of eyes on the code, though.

File: pyairtable/orm/model.py [hunk 1]
```diff
@@ -209,7 +209,8 @@ def save(self) -> bool:
         """
         table = self.get_table()
         record = self.to_record()
-        fields = record["fields"]
+        _fndm = self._field_name_descriptor_map()
```
Comment (NicoHood) [inline pyairtable/orm/model.py:212]: I personally avoid such abbreviations in variables. Of course, you can the the name on the right hand side, but it leads to abuse in further commits. Also I'd like to have a comment on what the loop does. Maybe it makes sense to split this one liner into multiple lines but explain it better.

File: tests/test_orm_fields.py [hunk 2]
```diff
@@ -27,6 +27,15 @@ class T:
     assert t.email == "x@x.com"
 
 
+def test_field_read_only():
+    class T:
+        name = f.Field("Name", read_only=True)
+
+    t = T()
+    t.name = "y"
+    assert t.name is None
```
Comment (NicoHood) [inline tests/test_orm_fields.py:36]: what does this test do? shouldnt setting the same rather throw an exception?

File: tests/test_orm.py [hunk 3]
```diff
@@ -52,6 +52,7 @@ class Contact(Model):
         is_registered = f.CheckboxField("Registered")
         link = f.LinkField("Link", Address, lazy=True)
         birthday = f.DateField("Birthday")
+        photo = f.Field("Photo", read_only=True)
```
Comment (NicoHood) [inline tests/test_orm.py:55]: Why is this just a "Field" and no "TextField"?

File: tests/test_orm.py [hunk 4]
```diff
@@ -64,20 +65,24 @@ class Meta:
         email="gui@gui.com",
         is_registered=True,
         birthday=datetime(2020, 12, 12).date(),
+        photo="{json: photo}"
```
Comment (NicoHood) [inline tests/test_orm.py:68]: what is this line meant to? It seems like invalid json

File: docs/source/orm.rst [hunk 5]
```diff
@@ -23,6 +23,7 @@ Fields
 
 .. automodule:: pyairtable.orm.fields
     :members:
+    :undoc-members:
```
Comment (NicoHood) [inline docs/source/orm.rst:26]: why was this added?

Comment (aleafa) [reply docs/source/orm.rst:26]: I think it should compile more robust ORM documentation. It's not a necessary addition by any means

File: tests/test_orm.py (hunk 4 above)
Comment (aleafa) [reply tests/test_orm.py:68]: It's filler text for a read-only field. Syntax shouldn't matter.

File: tests/test_orm.py (hunk 3 above)
Comment (aleafa) [reply tests/test_orm.py:55]: It's simulating a generic Field, not a TextField. Does that result in an unreliable test?

File: tests/test_orm_fields.py (hunk 2 above)
Comment (aleafa) [reply tests/test_orm_fields.py:36]: Oh, I think this test is based on a prior iteration of the code. It should be updated.

File: tests/test_orm.py (hunk 4 above)
Comment (NicoHood) [reply tests/test_orm.py:68]: as it is confusing, I'd set it to something less confusing like "Test Text" etc.

Comment (aleafa) [inline tests/test_orm.py:68]:
```suggestion
        photo="field data"
```

File: pyairtable/orm/model.py (hunk 1 above)
Comment (aleafa) [reply pyairtable/orm/model.py:212]: Sure. How about this?

```suggestion
        # Filter fields by read_only attribute so we only set values for writable fields
        map_ = self._field_name_descriptor_map()
```

File: pyairtable/orm/model.py [hunk 6]
```diff
@@ -209,7 +209,8 @@ def save(self) -> bool:
...
+        fields = {f: v for f, v in record["fields"].items() if not _fndm[f].read_only}
```
Comment (aleafa) [inline pyairtable/orm/model.py:213]:
```suggestion
        fields = {f: v for f, v in record["fields"].items() if not map_[f].read_only}
```

**Comment by gtalarico** [comment]
Sorry for the delay here @larsakerson will review this and incorporate it

**Comment by larsakerson** [comment]
My apologies. I've incorporated the suggestions; let me know if anything else is needed here, @gtalarico
